ttnn/cpp/ttnn/operations/moreh/moreh_sgd/device/moreh_sgd_device_operation.cpp

```diff
@@ -0,0 +1,118 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_sgd_device_operation.hpp"
+
+#include <cstdint>
+
+#include "tt_dnn/op_library/moreh_helper_functions.hpp"
+#include "ttnn/tensor/tensor.hpp"
+
+namespace ttnn::operations::moreh::moreh_sgd {
+void MorehSgdOperation::validate_inputs(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    auto& params_in = tensor_args.param_in;
+    auto& grad = tensor_args.grad;
+
+    tt::operations::primary::check_tensor(params_in, "moreh_sgd", "params_in");
+    tt::operations::primary::check_tensor(grad, "moreh_sgd", "grad");
+
+    if (tensor_args.momentum_buffer_in) {
+        tt::operations::primary::check_tensor(*tensor_args.momentum_buffer_in, "moreh_sgd", "momentum_buffer_in");
+    }
+
+    const auto& params_out = tensor_args.output_tensors.at(0);
+
+    if (params_out.has_value()) {
+        tt::operations::primary::check_tensor(params_out.value(), "moreh_sgd", "params_out");
+    }
+
+    if (tensor_args.output_tensors.at(1).has_value()) {
+        tt::operations::primary::check_tensor(
+            tensor_args.output_tensors.at(1).value(), "moreh_sgd", "momentum_buffer_out");
+    }
+}
+
+MorehSgdOperation::program_factory_t MorehSgdOperation::select_program_factory(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    // For now we litteraly don't care and return a single factory. Whatever
+    return ProgramFactory{};
```

Comment:
Remove comment

Reply:
resolved in [dc2ce28](https://github.com/tenstorrent/tt-metal/pull/12378/commits/dc2ce28ff17ddff2b33f87920282e226d4fec218)